cdc/api/middleware/middleware.go

```diff
@@ -99,21 +100,13 @@ func ForwardToControllerMiddleware(p capture.Capture) gin.HandlerFunc {
 // ForwardToChangefeedOwnerMiddleware forward a request to controller if current server
 // is not the changefeed owner, or handle it locally.
 func ForwardToChangefeedOwnerMiddleware(p capture.Capture,
-	changefeedID func(ctx *gin.Context) model.ChangeFeedID,
+	changefeedIDFunc func(ctx *gin.Context) model.ChangeFeedID,
 ) gin.HandlerFunc {
 	return func(ctx *gin.Context) {
-		// currently not only controller capture has the owner, remove this check in the future
-		if p.StatusProvider() != nil {
-			ok, err := p.StatusProvider().IsChangefeedOwner(ctx, changefeedID(ctx))
-			if err != nil {
-				_ = ctx.Error(err)
-				return
-			}
-			// this capture is the changefeed owner's capture, handle this request directly
-			if ok {
-				ctx.Next()
-				return
-			}
+		changefeedID := changefeedIDFunc(ctx)
+		// check if this capture is the changefeed owner
+		if handleRequestIfIsChnagefeedOwner(ctx, p, changefeedID) {
+			return
 		}
 
 		// forward to the controller to find the changefeed owner capture
```

Comment:
Whether it is possible to directly forward to the node where the owner resides?

Reply:
no,  I think only the controller know where is the changefeed owner